### PR TITLE
add transformState solution

### PR DESCRIPTION
**Pull request by DidenkoDenis306**

### CODE REVIEW

File: src/transformState.js [hunk 1]
```diff
@@ -5,7 +5,28 @@
...
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (true) {
+      case (action.type === 'addProperties'):
+        Object.assign(state, action.extraData);
+        break;
+
+      case (action.type === 'removeProperties'):
+        for (const key of action.keysToRemove) {
```
Comment (33Orange) [inline src/transformState.js:15]: Use array methods here instead nested loop. `Nested loops === EVIL`

File: src/transformState.js [hunk 2]
```diff
@@ -5,7 +5,28 @@
...
+        break;
+
+      case (action.type === 'clear'):
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+
+      default:
+        return null;
```
Comment (33Orange) [inline src/transformState.js]: [CODE STYLE]: switch/case should always have default case for error handling.

**Review by 33Orange — CHANGES_REQUESTED** [review]
The code works, but doesn't match several items in the checklist.

**Review by Anna-Tkachenko — APPROVED** [review]
⭐️

**Pull request closed without merging**